lxd/cluster/recover.go

```diff
@@ -538,3 +539,87 @@ func unpackTarball(tarballPath string, destRoot string) error {
 
 	return nil
 }
+
+// createTarball creates tarball at tarballPath, rooted at rootDir and including
+// all files in walkDir except those paths found in excludeFiles.
+// walkDir and excludeFiles elements are relative to rootDir.
+func createTarball(tarballPath string, rootDir string, walkDir string, excludeFiles []string) error {
```

Comment:
is this a copy + edit of an existing function?

Reply:
[Yes](https://github.com/canonical/microcluster/blob/v3/internal/recover/recover.go#L541)... I can drop this in `shared/` and import it over there if you'd prefer.

Reply:
no need to put it in shared, but if you can avoid duplicating the function and instead expand the scope of the existing function that would be great.

Reply:
oh wait, its from microcluster :)

Ah ok then its fine as is :)